jdatetime/__init__.py

```diff
@@ -1187,15 +1211,19 @@ def dst(self):
             return self.tzinfo.dst(self)
         return None
 
-    def isoformat(self):
+    def isoformat(self, sep='T', timespec='auto'):
         """[sep] -> string in ISO 8601 format, YYYY-MM-DDTHH:MM:SS[.mmmmmm][+HH:MM]."""
-        mil = self.strftime("%f")
-        if int(mil) == 0:
-            mil = ""
-        else:
-            mil = "." + mil
+        if not isinstance(sep, str) or len(sep) != 1:
+            raise TypeError('isoformat() argument 1 must be a unicode character, not str')
+
         tz = self.strftime("%z")
-        return self.strftime("%Y-%m-%dT%H:%M:%S") + "%s%s" % (mil, tz)
+
+        date_ = self.strftime("%Y-%m-%d")
+        time_ = _format_time(self.hour, self.minute, self.second,
+                             self.microsecond, timespec)
+
+        result = '%s%c%s%s' % (date_, sep, time_, tz)
```

Comment:
We could omit `result` here and just `return`. Also it would be great if we use new formatting style in any new code, and whenever changing existing code, we could replace the old style using `%`.

Reply:
I'm gonna try using f-style formatting, but let me know if that's now what you meant

Reply:
another option is `.format()`

Reply:
using `.format()` is what I'm suggesting.